### DISCUSSION TITLE
Why a new action and not a fork / move of an existing one?

### DISCUSSION THREAD

**Discussion by rajbos** [actions/create-github-app-token#39] [Q&A] [answered]
Hi folks, I wanted to reach out to you on this action, seems like this is duplicating the effort of @peter-murray's [excellent Action](https://github.com/peter-murray/workflow-application-token-action) for just the same thing. I'd either expect a move of Peter's repo or a fork, instead of a rewrite. Peter's action has been the go-to for several years now, and already has features like:
- only request a token for specific scopes
- request a token for a different organization
- token in another instance (GHES to GHEC for example, and any combination of the two)

Last time I checked Peter still works at GitHub, so communication should flow for this :-D.

**Comment by gr2m** [comment]
We weren't aware of Peter's action 😞 

The reason why we created a new one is that it's very simple and mostly uses @octokit dependencies which I created and GitHub maintains.

We are happy to accept outside contributions for the missing features we already identified, and also plan to contribute improvements back to actions that are in use today. For example Peter's action does not invalidate tokens in the post step yet.

**Comment by parkerbxyz** [comment, marked as answer]
This _was_ a move of an existing repo, just not Peter's version. This repo was previously [gr2m/github-app-token-action](https://github.com/gr2m/github-app-token-action).

There are several other GitHub Actions out there that serve the same purpose, but have different implementations:

- [tibdex/github-app-token](https://github.com/tibdex/github-app-token)
- [peter-murray/workflow-application-token-action](https://github.com/peter-murray/workflow-application-token-action)
- [getsentry/action-github-app-token](https://github.com/getsentry/action-github-app-token)
- [microsoftgraph/get-app-token](https://github.com/microsoftgraph/get-app-token)
- [wow-actions/use-app-token](https://github.com/wow-actions/use-app-token)
- [navikt/github-app-token-generator](https://github.com/navikt/github-app-token-generator)
- [mercari/github-app-token-generator](https://github.com/mercari/github-app-token-generator)
- [jnwng/github-app-installation-token-action](https://github.com/jnwng/github-app-installation-token-action)
- [And many more...](https://github.com/marketplace?category=&page=1&q=github+app+token+sort:popularity-desc&query=github+app+token+sort:popularity-desc&type=actions&verification=)
